core/src/test/java/org/elasticsearch/action/main/MainResponseTests.java

```diff
@@ -51,6 +53,20 @@ public static MainResponse createTestItem() {
         return new MainResponse(nodeName, version, clusterName, clusterUuid , build, available);
     }
 
+    public void testSerialization() throws IOException {
+        final MainResponse mainResponse = createTestItem();
+        BytesStreamOutput streamOutput = new BytesStreamOutput();
+        mainResponse.writeTo(streamOutput);
+        final MainResponse serialized = new MainResponse();
+        serialized.readFrom(streamOutput.bytes().streamInput());
+
+        assertThat(serialized.getNodeName(), equalTo(mainResponse.getNodeName()));
+        assertThat(serialized.getClusterName(), equalTo(mainResponse.getClusterName()));
+        assertThat(serialized.getBuild(), equalTo(mainResponse.getBuild()));
+        assertThat(serialized.isAvailable(), equalTo(mainResponse.isAvailable()));
+        assertThat(serialized.getVersion(), equalTo(mainResponse.getVersion()));
```

Comment:
good one, I am also looking into migrating the whole class to some other existing base class like `AbstractStreamableXContentTestCase`, just found out it exists :)